tests/NuGet.Services.AzureSearch.Tests/SearchService/SearchTextBuilderFacts.cs

```diff
@@ -0,0 +1,170 @@
+        {
+            [Theory]
+            [MemberData(nameof(CommonAzureSearchQueryData))]
+            public void GeneratesAzureSearchQuery(string input, string expected)
+            {
+                var actual = _target.V2Search(new V2SearchRequest { Query = input });
+
+                Assert.Equal(expected, actual);
+            }
+
+            // TODO: id and packageId query fields should map to different fields with different analyzers
+            // See: https://github.com/NuGet/NuGetGallery/issues/6920
+            // See: https://github.com/NuGet/NuGetGallery/issues/6922
+            [Theory]
+            [InlineData(false, "packageId:hello")]
+            [InlineData(true, "packageId:hello")]
+            public void WhenLuceneQuery_TreatsLeadingIdAsPackageId(bool luceneQuery, string expected)
+            {
+                var actual = _target.V2Search(new V2SearchRequest
+                {
+                    Query = "id:hello",
+                    LuceneQuery = luceneQuery,
+                });
+
+                Assert.Equal(expected, actual);
+            }
+        }
+
+        public class V3Search : FactsBase
+        {
+            [Theory]
+            [MemberData(nameof(CommonAzureSearchQueryData))]
+            public void GeneratesAzureSearchQuery(string input, string expected)
+            {
+                var actual = _target.V3Search(new V3SearchRequest { Query = input });
+
+                Assert.Equal(expected, actual);
+            }
+        }
+
+        public class FactsBase
+        {
+            protected readonly SearchTextBuilder _target;
+
+            public FactsBase()
+            {
+                _target = new SearchTextBuilder();
+            }
+
+            public static IEnumerable<object[]> CommonAzureSearchQueryData()
+            {
+                // Map of inputs to expected output
+                var data = new Dictionary<string, string>
+                {
+                    { "", "*" },
+                    { " ", "*" },
+
+                    // TODO: id and packageId query fields should map to different fields with different analyzers
+                    // See: https://github.com/NuGet/NuGetGallery/issues/6920
+                    // See: https://github.com/NuGet/NuGetGallery/issues/6922
+                    { "id:test", "packageId:test" },
+                    { "packageId:json", "packageId:json" },
+                    { "version:1.0.0", "normalizedVersion:1.0.0" },
+                    { "title:hello", "title:hello" },
+                    { "description:hello", "description:hello" },
+                    { "tag:hi", "tags:hi" },
+                    { "tags:foo", "tags:foo" },
+                    { "author:bob", "authors:bob" },
+                    { "authors:billy", "authors:billy" },
+                    { "summary:test", "summary:test" },
+                    { "owner:goat", "owners:goat" },
+                    { "owners:nugget", "owners:nugget" },
```

Comment:
Every field and non-scoped terms should have assertions that we don't change the casing. In general I think we should depend on the analyzer to do as much as possible

Reply:
Added a test